cpp/include/cudf/datetime.hpp

```diff
@@ -133,6 +133,48 @@ std::unique_ptr<cudf::column> extract_second(
   cudf::column_view const& column,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief  Extracts millisecond from any date time type and returns an int16_t
+ * cudf::column.
+ *
+ * @param column cudf::column_view of the input datetime values
+ * @param mr Device memory resource used to allocate device memory of the returned column
+ *
+ * @returns cudf::column of the extracted int16_t milliseconds
+ * @throw cudf::logic_error if input column datatype is not TIMESTAMP
+ */
+std::unique_ptr<cudf::column> extract_millisecond(
+  cudf::column_view const& column,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief  Extracts microsecond from any date time type and returns an int16_t
+ * cudf::column.
```

Comment:
This is a bit odd. The code appears to return not microseconds but a microsecond fraction. So for a duration of `1.234567890` it returns `567`  and not `234567` which is the microsecond portion.
The millisecond function returns `234` which makes sense but nanosecond returns `890` when I would expect to see `234567890`.
Is this what is intended? If so, I think we need more documentation here. I would also suggest changing the name. Perhaps `extract_microsecond_fraction` or something like that.

Reply:
Yes, we would want to extract a fraction here. I'll change the names.

Reply:
Renamed the method here.